src/utils/files.ts

```diff
@@ -1,24 +1,69 @@
-import axios from "axios";
+import axios, { AxiosRequestConfig } from "axios";
 import logger from '../logger';
 
-export async function downloadFileFromUrl(url: string): Promise<ArrayBuffer> {
-	logger.info(`Downloading file from ${url}`);
+// export async function downloadFileFromUrl(url: string): Promise<ArrayBuffer> {
+// 	logger.info(`Downloading file from ${url}`);
 
-	try {
-		const response = await axios({
-			method: "get",
-			url,
-			responseType: "arraybuffer"
-		});
+// 	try {
+// 		const response = await axios({
+// 			method: "get",
+// 			url,
+// 			responseType: "arraybuffer"
+// 		});
 
-		logger.info(`File downloaded successfully from ${url}`);
+// 		logger.info(`File downloaded successfully from ${url}`);
 		
-		return response.data;
+// 		return response.data;
+// 	} catch (err) {
+// 		logger.error(`Failed to download file from ${url}: ${err}`);
+// 		throw err;
+// 	}
+// }
+
+//I have commented the old function that does not cater for Auth
+
+export async function downloadFileFromUrl(
+	url: string,
+	auth?: {
+	  username: string;
+	  password: string;
+	} | {
+	  token: string;
+	  tokenType?: string; 
+	}
+  ): Promise<ArrayBuffer> {
+	logger.info(`Downloading file from ${url}`);
+  
+	try {
+	  const config: AxiosRequestConfig = {
+		method: "get",
+		url,
+		responseType: "arraybuffer"
+	  };
+  
+	  if (auth) {
+		if ('username' in auth) {
```

Comment:
```suggestion
                if (auth.username) {
```

Think it's best to check if a value exists instead of just checking if a key exists.